dsl/src/simple_function/task.rs

```diff
@@ -140,9 +140,14 @@ impl Fiber {
   // Runs until finished and gets the resutl or until parked for awaiting async results
   pub fn run(&mut self) -> RunResult {
     loop {
-      let Some(head) = self.stack.pop() else {
-        panic!("no way there will be no elements. Can happen only on empty one")
-      };
+      let head_opt = self.stack.pop();
+      if head_opt.is_none() {
+        // Empty stack indicates completion for void-returning functions.
```

Comment:
This does not sound right!

The stack should be empty when the top-level returning function is a `void` (as it should be, unless we have "return codes" for fibers), but the reverse condition is not true I think.